src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -252,9 +239,18 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                                 }
                             }
                         }
-                        catch (DiagnosticsClientException e)
+                        catch (Exception ex) when (ex is DiagnosticsClientException || ex is System.UnauthorizedAccessException)
```

Comment:
```suggestion
                        catch (DiagnosticsClientException e)
                        {
                            Console.Error.WriteLine($"Unable to start a tracing session: {e.ToString()}");
                            return ReturnCode.SessionCreationError;
                        }
                        catch(UnauthorizedAcessException e)
                        {
                            Console.Error.WriteLine($"dotnet-trace does not have permission to access the specified app: {e.GetType()}")
                            return ReturnCode.SessionCreationError;
                        }
```
The way you wrote it would also work, this way is just a more common style in C# for handling multiple Exception types.